### ISSUE TITLE
AWS::Timestream::InfluxDBInstance marked as in mantainance incorrectly

### ISSUE THREAD

**Issue by camillacolser** [aws-cloudformation/cfn-lint#4435]
### CloudFormation Lint Version

1.48.0

### What operating system are you using?

Mac

### Describe the bug

Double checking the error for the resource `AWS::Timestream::InfluxDBInstance` on https://docs.aws.amazon.com/general/latest/gr/maintenance_services.html, it seems that the service is not in mantainance mode, as suggested by the linter.

Error:
```
W3697 Resource type 'AWS::Timestream::InfluxDBInstance' is from a service in maintenance mode since 2025-06-20. Consider migrating to an alternative
```

### Expected behavior

No error for `AWS::Timestream::InfluxDBInstance`

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Description: >
  TimeStream for InfluxDB configuration
Parameters:
  Project:
    Type: String
    AllowedValues: [ project ]
  Environment:
    Type: String
    Default: beta
    AllowedValues: [ beta, prod ]
  DbInstanceType:
    Type: String
    Default: db.influx.medium
  DeploymentType:
    Type: String
    Default: SINGLE_AZ
    AllowedValues: [ SINGLE_AZ, WITH_MULTIAZ_STANDBY ]
  AllocatedStorage:
    Type: Number
    Default: 20
    MinValue: 20

Resources:
  TimestreamInfluxDB:
    Type: AWS::Timestream::InfluxDBInstance
    Properties:
      Name: !Ref Project
      DbInstanceType: !Ref DbInstanceType
      AllocatedStorage: !Ref AllocatedStorage
      Username: admin
      Password: !Sub "{{resolve:secretsmanager:/${Environment}/${Project}/timestream-influxdb/application:SecretString:password}}"
      Organization: !Ref Project
      Bucket: catalog_api
      VpcSubnetIds:
        - Fn::ImportValue: !Sub "${Environment}-private-subnet-a"
        - Fn::ImportValue: !Sub "${Environment}-private-subnet-b"
        - Fn::ImportValue: !Sub "${Environment}-private-subnet-c"
      VpcSecurityGroupIds:
        - !Ref EC2SecurityGroup
      DeploymentType: !Ref DeploymentType
      PubliclyAccessible: false
      NetworkType: IPV4
      Port: 8086

  TimestreamInfluxDBSecret:
    Type: AWS::SecretsManager::Secret
    Properties:
      Name: !Sub "/${Environment}/${Project}/timestream-influxdb/application"
      Description: 'Catalog access to Timestream for InfluxDB credentials'
      GenerateSecretString:
        SecretStringTemplate: !Sub '{"org": "${Project}"}'
        GenerateStringKey: 'password'
        PasswordLength: 16
        ExcludePunctuation: true

  EC2SecurityGroup:
    Type: AWS::EC2::SecurityGroup
    Properties:
      GroupDescription: !Sub "${Environment}-${Project}-timestream-influxdb"
      GroupName: !Sub "${Environment}-${Project}-timestream-influxdb-sg"
      VpcId:
        Fn::ImportValue: !Sub "${Environment}-vpc"
      SecurityGroupIngress:
        - IpProtocol: tcp
          SourceSecurityGroupId:
            Fn::ImportValue: catalog-workers-sg
          FromPort: 8086
          ToPort: 8086
        - IpProtocol: tcp
          SourceSecurityGroupId:
            Fn::ImportValue: catalog-api-internal-sg
          FromPort: 8086
          ToPort: 8086
        - IpProtocol: tcp
          SourceSecurityGroupId:
            Fn::ImportValue: catalog-batch-sg
          FromPort: 8086
          ToPort: 8086
        - IpProtocol: tcp
          SourceSecurityGroupId:
            Fn::ImportValue: catalog-api-sg
          FromPort: 8086
          ToPort: 8086
        - IpProtocol: tcp
          SourceSecurityGroupId:
            Fn::ImportValue: catalog-data-product-sg
          FromPort: 8086
          ToPort: 8086
        - IpProtocol: tcp
          SourceSecurityGroupId:
            Fn::ImportValue: !Sub VPN-${Environment}-sg
          FromPort: 8086
          ToPort: 8086
      SecurityGroupEgress:
        - CidrIp: 0.0.0.0/0
          IpProtocol: '-1'


Outputs:
  EC2SecurityGroup:
    Description: Security Group assigned to timestream for influxdb
    Value: !Ref EC2SecurityGroup
    Export:
      Name: !Sub "${Project}-timestream-influxdb-sg"

  TimestreamEndpoint:
    Description: Timestream endpoint
    Value: !GetAtt TimestreamInfluxDB.Endpoint
    Export:
      Name: !Sub "${Project}-timestream-influxdb-endpoint"

```
